### PR TITLE
[SP-4720] Backport of PDI-17646 - ConcurrentModificationException in …

### PR DESCRIPTION
**Pull request by smmribeiro**
…AbstractMeta (7.1 Suite)

In 7.1 there's no "currentDirectoryChangedListeners" so I did not backport any of the related changes.

@LeonardoCoelho71950 @pentaho-lmartins @mbatchelor

### CODE REVIEW

File: engine/src/org/pentaho/di/base/AbstractMeta.java [hunk 1]
```diff
@@ -50,13 +48,16 @@
 import org.pentaho.di.core.logging.LogChannel;
 import org.pentaho.di.core.logging.LogLevel;
 import org.pentaho.di.core.logging.LoggingObjectInterface;
+import org.pentaho.di.core.osgi.api.MetastoreLocatorOsgi;
```
Comment (pentaho-lmartins) [inline engine/src/org/pentaho/di/base/AbstractMeta.java]: Wingman reports:
```
AbstractMeta.java:51: error: package org.pentaho.di.core.osgi.api does not exist
[javac] import org.pentaho.di.core.osgi.api.MetastoreLocatorOsgi;
```

File: engine/src/org/pentaho/di/base/AbstractMeta.java [hunk 2]
```diff
@@ -50,13 +48,16 @@
...
+import org.pentaho.di.core.osgi.api.NamedClusterServiceOsgi;
```
Comment (pentaho-lmartins) [inline engine/src/org/pentaho/di/base/AbstractMeta.java]: Wingman reports:
```
AbstractMeta.java:52: error: package org.pentaho.di.core.osgi.api does not exist
[javac] import org.pentaho.di.core.osgi.api.NamedClusterServiceOsgi;
```

**Comment by smmribeiro** [comment]
The newly included test is failing (but it passed successfully in master and 8.1):

```
[junit] Exception in thread "main" java.lang.OutOfMemoryError: Java heap space
[junit] Running org.pentaho.di.base.AbstractMetaTest
[junit] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 sec
[junit] Test org.pentaho.di.base.AbstractMetaTest FAILED (crashed)
```
I'm running wingman again to see if it passes as is...

**Comment by smmribeiro** [comment]
It was necessary to set a higher value for the "junit.maxmemory" parameter for the test to pass (the other way to make the test pass was to lower the number of iterations performed - and that would be kind of cheating ;-)

**Pull request merged**